pkg/gooq/function.go

```diff
@@ -104,6 +104,42 @@ func (expr *expressionFunctionImpl) Render(
 // [Good First Issue][Help Wanted] TODO: implement remaining functions (not operators)
 ///////////////////////////////////////////////////////////////////////////////
 
+func Ascii(
+	input Expression,
```

Comment:
According to the [documentation](https://www.postgresql.org/docs/11/functions-string.html#FUNCTIONS-STRING-OTHER), ASCII has the following signature:

```
ascii(string) -> int
```

We probably want stricter typing to ensure the user cannot pass in values that will lead to a runtime error. For example, the following test will compile and pass:

```go
	{
		Constructed:  Select(Ascii(Int64(5))),
		ExpectedStmt: "SELECT ASCII($1)",
	},
```

However, running `SELECT ASCII(5)` in Postgres (`postres:latest` on Docker) produces the following results:

```
postgres=# SELECT(ASCII(5));
ERROR:  function ascii(integer) does not exist
LINE 1: SELECT(ASCII(5));
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

We probably want `input` to be a `StringExpression` and the return type to be `NumericExpression`.

There are similar problems with `BTrim` and `LTrim`: we're not fully leveraging the types.

Other than the typing issue, everything seems fine though :+1:

Reply:
I agree for the stricter types, but im not sure how to go around it. Would it be by implementing function types that return numeric/string expressions?

Reply:
Sorry for misleading you, but I just realized that we don't seem to have a way of expressing return types for function types: https://github.com/lumina-tech/gOOQ/blame/master/pkg/gooq/function.go